tests/performance_estimation/CBPE/test_cbpe.py

```diff
@@ -751,3 +786,67 @@ def test_cbpe_fitting_does_not_generate_error_when_single_class_present():
         business_value_matrix=[[1, -1], [-1, 1]],
     )
     sut.fit(ref_df)
+
+
+def test_cbpe_returns_distinct_but_consistent_results_when_reused_noopcal(  # noqa: D103
+    binary_classification_data
+):
+    reference, analysis = binary_classification_data
+
+    sut = CBPE(
+        # timestamp_column_name='timestamp',
+        chunk_size=50_000,
+        y_true='work_home_actual',
+        y_pred='y_pred',
+        y_pred_proba='y_pred_proba',
+        metrics=['roc_auc'],
+        problem_type='classification_binary',
+        calibrator=NoopCalibrator()
+    )
+    sut.fit(reference)
+    result1 = sut.estimate(analysis)
+    result2 = sut.estimate(analysis)
+    assert result1 is not result2
+    pd.testing.assert_frame_equal(result1.to_df(), result2.to_df())
+
+
+def test_oo_behavior_bc_perf_est(binary_classification_data):  # noqa: D103
+    reference, monitored = binary_classification_data
+    reference2 = reference.copy(deep=True)
+    monitored2 = monitored.copy(deep=True)
+    estimator = CBPE(
+        # timestamp_column_name='timestamp',
+        chunk_size=50_000,
+        y_true='work_home_actual',
+        y_pred='y_pred',
+        y_pred_proba='y_pred_proba',
+        metrics=['roc_auc', 'f1'],
+        problem_type='classification_binary',
+    )
+    estimator.fit(reference2)
+    results = estimator.estimate(monitored2)  # noqa: F841
+    pd.testing.assert_frame_equal(monitored, monitored2)
+    pd.testing.assert_frame_equal(reference, reference2)
+
+
+def test_oo_behavior_mc_perf_est(multiclass_classification_data):  # noqa: D103
```

Comment:
Name could be a bit more descriptive, as this is a weird thing to test.

Maybe something like `test_input_dataframes_are_not_altered_by_calculator`